pkg/storage/storage.go

```diff
@@ -12,14 +12,19 @@ type ExternalStorage interface {
 	// Read storage file
 	Read(name string) ([]byte, error)
 	// FileExists return true if file exists
-	FileExists(name string) bool
+	FileExists(name string) (bool, error)
 }
 
 // Create creates ExternalStorage
 func Create(backend *backup.StorageBackend) (ExternalStorage, error) {
 	switch backend := backend.Backend.(type) {
 	case *backup.StorageBackend_Local:
 		return newLocalStorage(backend.Local.Path)
+	case *backup.StorageBackend_S3:
+		if backend.S3 == nil {
+			return nil, errors.Errorf("no s3 config in %#v", backend)
```

Comment:
Would this expose the secrets in the log?

Reply:
Since backend.S3 is nil, there should be no secret in the log. Anyway, I will remove the backend.